app/modules/entities/components/editor/select_author_role.svelte

```diff
@@ -0,0 +1,78 @@
+<script>
+  import { isNonEmptyClaimValue } from '#entities/components/editor/lib/editors_helpers'
+  import { getWorkPreferredAuthorRolesProperties } from '#entities/lib/editor/properties_per_subtype'
+  import preq from '#lib/preq'
+  import { onChange } from '#lib/svelte/svelte'
+  import { I18n, i18n } from '#user/lib/i18n'
+  import { uniq, without } from 'underscore'
+
+  export let entity, property, value
+
+  let rolesProperties
+  function setRolesProperties () {
+    rolesProperties = getWorkPreferredAuthorRolesProperties(entity)
+    if (!rolesProperties.includes(property)) {
+      rolesProperties = [ property, ...rolesProperties ]
+    }
+    if (rolesProperties.length === 1 && rolesProperties[0] === property) {
+      rolesProperties = null
+    }
+  }
+
+  $: onChange(entity, setRolesProperties)
+
+  let currentRoleProperty = property
+
+  async function onRolePropertyChange () {
+    if (currentRoleProperty !== property) {
+      entity.claims[property] = without(entity.claims[property], value)
+      entity.claims[currentRoleProperty] = entity.claims[currentRoleProperty] || []
+      entity.claims[currentRoleProperty] = uniq(entity.claims[currentRoleProperty].concat([ value ]))
+      // If entity.uri is undefined, we are manipulating a not-yet-created entity
+      if (entity.uri) {
+        await preq.put(app.API.entities.claims.update, {
+          uri: entity.uri,
+          property,
+          'old-value': value,
+          'new-value': null,
+        })
+        await preq.put(app.API.entities.claims.update, {
+          uri: entity.uri,
+          property: currentRoleProperty,
+          'old-value': null,
+          'new-value': value,
+        })
+      }
+    }
+  }
+
+  $: onChange(currentRoleProperty, onRolePropertyChange)
+</script>
+
+{#if rolesProperties && isNonEmptyClaimValue(value)}
+  <label>
+    {i18n('Change author role')}
+    <select bind:value={currentRoleProperty}>
+      {#each rolesProperties as roleProperty}
+        {@const disabled = roleProperty !== property && entity.claims[roleProperty]?.includes(value)}
+        <option
+          value={roleProperty}
+          {disabled}
+          title={disabled ? i18n('This author already has that role') : ''}
```

Comment:
`author` string can be ambiguous here, since the `role` can have `author` value. Then, when you hover on disabled `author` role, it feels redundant, one may say "This author already has that author role"). We could find a way to be more abstract: 

```suggestion
          title={disabled ? i18n('This person already has that role') : ''}
```
but are organizations persons?

Even more abstract:
```suggestion
          title={disabled ? i18n('They already has that role') : ''}
```

Reply:
Depending on the choice made between the [atlernatives above](https://github.com/inventaire/inventaire-client/pull/434#issuecomment-1791275470), we might have the possibility to keep "author" without it adding confusion

Reply:
Change done in 03ebb42a8. I think that we will have cases where author is still used as the generic term, while still covering more specific roles, for instance in the case of the Author layout, and I think it's ok